qt_ui/widgets/QTopPanel.py

```diff
@@ -78,14 +79,24 @@ def __init__(self, game_model: GameModel, sim_controller: SimController) -> None
         self.buttonBoxLayout.addWidget(self.transfers)
         self.buttonBox.setLayout(self.buttonBoxLayout)
 
+        self.simSpeedControls = SimSpeedControls(sim_controller)
+
         self.proceedBox = QGroupBox("Proceed")
         self.proceedBoxLayout = QHBoxLayout()
-        self.proceedBoxLayout.addLayout(SimSpeedControls(sim_controller))
+        self.proceedBoxLayout.addLayout(self.simSpeedControls)
         self.proceedBoxLayout.addLayout(MaxPlayerCount(self.game_model.ato_model))
         self.proceedBoxLayout.addWidget(self.passTurnButton)
         self.proceedBoxLayout.addWidget(self.proceedButton)
         self.proceedBox.setLayout(self.proceedBoxLayout)
 
+        self.controls = [
```

Comment:
I think this would be better inverted. That is, add a signal for game state (which might just be the playing/ended boolean, but maybe useful to have the full set of not loaded/turn 0/playing/over?) and move the logic for enable/disable into the components themselves rather than the owner.

WDYT?

Reply:
I think that would make things less readable. If I understood what you're saying, it would require connecting the `game_state_changed` signal to all of those controls separately. Let alone the fact that we need to check whether or not the campaign has started, because that affects the state of some of the controls as well.

Either way, I'll look into the whole refactor thing together with the checks for the game state that are responsible for emitting the signals.

Reply:
I still think it's best to handle stuff here, however the logic in `setGame` is better off in a separate method that's linked to the `game_state_changed` signal, which I already did on my end 👍